examples/amm/amm_test.py

```diff
@@ -673,20 +735,73 @@ def burn(
                     ConstantProductAMMErrors.AssetPoolIncorrect,
                     override_axfer_asset(burn(), "pool_xfer", a_asset),
                 ),
-                # (ConstantProductAMMErrors.SenderInvalid, burn(app_client=fake_client)), # Not working
             ],
+        ) + cases(
+            ConstantProductAMM.burn,
+            [(ConstantProductAMMErrors.SenderInvalid, burn())],
+            fake_client,
         )
 
         return well_formed_burn + valid_pool_xfer
 
-    # TODO: rest of them
+    def swap_cases():
+        def swap(
+            app_client: client.ApplicationClient = creator_app_client,
+            assets: tuple[int, int] = assets,
+            swap_amt: int = 1,
+            swap_asset: int = a_asset,
+        ):
+            return build_swap_transaction(app_client, assets, swap_amt, swap_asset)
+
+        well_formed_swap = cases(
+            ConstantProductAMM.swap,
+            [
+                (
+                    ConstantProductAMMErrors.AssetAIncorrect,
+                    swap(assets=(b_asset, b_asset)),
+                ),
+                (
+                    ConstantProductAMMErrors.AssetBIncorrect,
+                    swap(assets=(a_asset, a_asset)),
+                ),
+            ],
+        )
+
+        valid_swap_xfer = cases(
+            ConstantProductAMM.swap,
+            [
+                (ConstantProductAMMErrors.AmountLessThanMinimum, swap(swap_amt=0)),
+                (
+                    ConstantProductAMMErrors.SendAmountTooLow,
+                    swap(swap_amt=1),
+                ),
+                (
+                    ConstantProductAMMErrors.AssetIdsIncorrect,
+                    swap(swap_amt=0, swap_asset=pool_asset),
+                ),
+            ],
+        ) + cases(
+            ConstantProductAMM.swap,
+            [(ConstantProductAMMErrors.SenderInvalid, swap())],
+            fake_client,
+        )
+
+        return well_formed_swap + valid_swap_xfer
 
     all_asserts: dict[int, ProgramAssertion] = creator_app_client.approval_asserts  # type: ignore[assignment]
-    for msg, method, kwargs in bootstrap_cases() + mint_cases() + burn_cases():
-        print(f"Testing {msg}")
+
+    print(f"Total asserts ({len(all_asserts)})")
```

Comment:
@barnjamin Minor and if you feel it's out of scope, I'm on-board to approve as is.  Let me know what you think?

I think it'd be nice to factor out 2 tests that use all test cases as input (naming can be modified):
* `test_asserts` - Confirms that each case raises an assertion.  Helps user understand when assertion test coverage regresses.
* `test_assert_coverage` - Fails when > 0 assertions are _not_ tested.  Helps user understand when assertion test coverage is partial.

I tried to do accomplish the goal locally, but I'm unable to make it work.  I think the idea is to return 
```   
        set_governor_cases()
        + bootstrap_cases()
        + mint_cases()
        + burn_cases()
        + swap_cases()
```

The challenge I ran into is combining `@pytest.mark.parameterize` with `@pytest.fixture`.  The parameterize input requires fixture values as input, and I was unable to find a way to make it work.  

I can also explain live if helpful.  I'm sure there's a way and it must be my lack of pytest awareness that's holding me up.